### PR TITLE
Reduce test run duration

### PR DESCRIPTION
**Pull request by Beau-Gosse-dev**
## Description
* Before this change, the tests would download the example solutions multiple times in a test run. Now, they are downloaded once and then copied to temp folders as needed.
* Reduces full-repository test run duration from around 10 minutes to around 3 minutes (on my local laptop at least)

## Supplemental testing
Ran command line against test solution

## Additional context
If we want, I can go further with this and create something like a lazy-loaded example solution factory that could potentially be used with CTA and other repositories. 

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs
Comment (Beau-Gosse-dev) [inline src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs:29]: What is our process for changing public APIs? I see this in included with the nuget package published by Codelyzer (and consumed by CTA). I didn't see this method used in CTA, but that's not to say it isn't used somewhere. Should I instead mark it as deprecated and we can remove it in a major package version change?

File: src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs
Comment (Beau-Gosse-dev) [inline src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs:37]: Another question on public API surfaces: Is it okay to put this here? It seems like a more robust solution (mostly copied from Microsoft documentation). I saw CTA is doing something similar on it's own. Should CTA be using this instead once/if it is published? CTA: https://github.com/aws/cta/blob/dcdf4dde2c6a105b0fe2115c3f275686999806b1/src/CTA.Rules.Config/Utils.cs#L216

File: src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs
Comment (jonlouie) [reply src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs:29]: No strict process. Instead, any files with meaningful public APIs require approval from Code Owners. 

In this case, I don't believe CreateDirectory() is used in other repos so this should be a safe change. Worst case scenario it will be a trivial refactor.

**Review by jonlouie — DISMISSED** [review]

File: src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs
Comment (jonlouie) [reply src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs:37]: Since CTA ports code and Codelyzer analyzes it, I have no strong opinion on where FileUtils should live. If we do end up sharing code, I agree it should be in Codelyzer since it's the lowest level dependency.

Comment (Eruanion) [reply src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs:37]: We may end up creating a support library project but as things currently stand I agree with John the lowest level dependency being Codelyzer should hold this for now.

**Review by Eruanion — APPROVED** [review]

**Comment by Beau-Gosse-dev** [comment]
Checks are failing due to failing test that should be fixed once this PR is merged: https://github.com/aws/codelyzer/pull/125

File: src/Codelyzer.sln.DotSettings
Comment (marknfawaz) [inline src/Codelyzer.sln.DotSettings:2]: Any reason this needs to be checked in?

**Review by marknfawaz — CHANGES_REQUESTED** [review]
Please update so the tests are passing. We might have a different version of the repo in our test projects, so either we add the commit being tested to the test projects or we ned to update our tests to make sure we are getting the correct outcome.

File: src/Codelyzer.sln.DotSettings
Comment (Beau-Gosse-dev) [reply src/Codelyzer.sln.DotSettings:2]: This is telling Resharper that the word "Codelyzer" isn't a typo. I got a resharper subscription when I joined and I assumed some other people were using it too. Would it be okay if we configure some solution-specific resharper settings like this in this file? If not, I'm happy to remove this and just keep the settings locally on my machine. 

This setting was just added through the UI popups clicking "add to dictionary -> Solution team-shared"

Comment (marknfawaz) [reply src/Codelyzer.sln.DotSettings:2]: It's a good idea to have this for sure. My only concern here is would this create any issues if someone doesn't have resharper installed?

Comment (Beau-Gosse-dev) [reply src/Codelyzer.sln.DotSettings:2]: @marknfawaz I’ve merged the latest main branch in with Jon’s fixes and that fixed the failing test. I also ran this on another machine without resharper and the dotsettings file didn’t interfere at all. Let me know if you have any other comments and if you are okay with the comments above that I posted about changing the public surface of the Utils. Jon and I will cut a new release for this once we get this merged into main.

**Review by jonlouie — APPROVED** [review]

**Pull request merged**
